src/reanimated2/animations.js

```diff
@@ -394,6 +394,7 @@ export function repeat(_nextAnimation, numberOfReps = 2, reverse = false) {
       animation.current = nextAnimation.current;
       if (finished) {
         animation.reps += 1;
+        callback();
```

Comment:
I think it could be good to pass `animation.current` to the callback. Sometimes reading sv value inside the callback does not give exact results. What do you think?

Reply:
What does sv stand for? For course, I wouldn't mind do this change.

Reply:
sv=shared value, sorry for using such shortcuts

Reply:
my point is let's say you have an example where you alternate between 100 and 200. With short periods if you'd like to use the current value somehow in the callbacks it could be wrong:
```
const mySV = useSharedValue(100) // this is gonna alternate between 100 and 200
mySV.value = repeat(/*...*/, () => { const flag = (mySV === 200); /* do something with the flag */ /* and this is going to sometimes give you a result of 200 but sometimes something close but not exact, like 198.34 */ })
```
instead
```
const mySV = useSharedValue(100) // this is gonna alternate between 100 and 200
mySV.value = repeat(/*...*/, (currentValue) => { const flag = (currentValue === 200); /* do something with the flag */ /* this is most likely going to be exact */ })
```

Reply:
I added the parameter to the callback.

Reply:
What does `defineAnimation` do exactly? And do we need it for every custom animations we build?

Reply:
it distinguishes style updates on JS/UI I think it should be used in order to prevent flickering

Reply:
Should it be exported for people who build custom animations?

Reply:
Maybe but I would consider exposing more easy-going tool for that